### PR TITLE
Add resource_logging_billing_account_sink resource

### PR DESCRIPTION
**Pull request by selmanj**
This adds support for `resource_logging_billing_account_sink`; the API looks identical to the one used for project-level sinks, so I opted to do some refactoring here to establish a common schema (there is also organizations and folder-level sinks coming soon).

### CODE REVIEW

File: google/resource_logging_billing_account_sink.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceLoggingBillingAccountSink() *schema.Resource {
+	schm := &schema.Resource{
+		Create: resourceLoggingBillingAccountSinkCreate,
+		Read:   resourceLoggingBillingAccountSinkRead,
+		Delete: resourceLoggingBillingAccountSinkDelete,
+		Update: resourceLoggingBillingAccountSinkUpdate,
+		Schema: resourceLoggingSinkSchema(),
```
Comment (rosbo) [inline google/resource_logging_billing_account_sink.go]: Import support? You have all the context you need fresh in your mind. Since this PR is long, I think it is a good idea to do it in a separate PR.

File: website/docs/r/logging_billing_account_sink.html.markdown [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the logging sink.
+
+* 'billing_account' - (Required) The billing account exported to the sink.
```
Comment (rosbo) [inline website/docs/r/logging_billing_account_sink.html.markdown]: use the same quote style: `` not `'`

File: google/resource_logging_sink.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+func flattenResourceLoggingSink(d *schema.ResourceData, sink *logging.LogSink) {
+	d.Set("name", sink.Name)
+	d.Set("destination", sink.Destination)
+	d.Set("filter", sink.Filter)
+	d.Set("writer_identity", sink.WriterIdentity)
+}
+
+func expandResourceloggingSinkForUpdate(d *schema.ResourceData) *logging.LogSink {
+	// Can only update destination/filter right now. Despite the method below using 'Patch', the API requires both
```
Comment (rosbo) [inline google/resource_logging_sink.go]: Capital `L` for logging.

File: google/resource_logging_billing_account_sink.go (hunk 1 above)
Comment (selmanj) [reply google/resource_logging_billing_account_sink.go]: Agreed! I actually started adding that but stopped as it got pretty complicated. My approach was modeled on yours

File: website/docs/r/logging_billing_account_sink.html.markdown (hunk 2 above)
Comment (selmanj) [reply website/docs/r/logging_billing_account_sink.html.markdown]: oops, fixed

File: google/resource_logging_sink.go (hunk 3 above)
Comment (selmanj) [reply google/resource_logging_sink.go]: oops, fixed
